platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorStyledTextView.kt

```diff
@@ -128,4 +134,38 @@ open class EditorStyledTextView : AppCompatTextView {
             }
         )
     }
+
+    override fun onTouchEvent(event: MotionEvent?): Boolean {
+        return when (event?.action) {
+            MotionEvent.ACTION_DOWN -> {
+                // Allow ACTION_UP to be called
+                super.onTouchEvent(event)
+                true
+            }
+            MotionEvent.ACTION_UP -> {
+                // Find selection matching the pointer coordinates
+                val offset = getOffsetForPosition(event.x, event.y)
+                // Look for clickable spans in that position
+                val spans = (text as? Spanned)?.getSpans(offset, offset, Any::class.java) ?: return false
```

Comment:
Nice idea! A few thoughts which may or may not not be issues:
- Have you tested if there are any conflicts using `setMovementMethod(LinkMovementMethod)` after this change?
- What happens if an app returns a custom clickable span in the mention display handler?
- Could this trigger if a user scrolls by dragging a link/pill/mention?

Reply:
> * Have you tested if there are any conflicts using `setMovementMethod(LinkMovementMethod)` after this change?

I think `LinkMovementMethod` needs the `TextView` to be selectable, which we don't allow at the moment, but I'd have to double check. Maybe instead of returning `true` in this case I should call the super function. Nice catch.

> * What happens if an app returns a custom clickable span in the mention display handler?

I think internally it would be represented by a `CustomMentionSpan`, which is a `ReplacementSpan` and can't contain a `ClickableSpan`. With `TextDisplay.Custom` you can just pass a `ReplacementSpan` if I'm not mistaken.

> * Could this trigger if a user scrolls by dragging a link/pill/mention?

Oh, nice catch. Maybe I should perform this same search in `ACTION_DOWN`, keep a reference to the selected spans and then check if I get the same ones in `ACTION_UP`.

Reply:
I couldn't add any automated tests for the 1st question, but I can confirm it works. I also added some new tests and made sure scrolling won't trigger a click (which apparently does happen with the default `ClickableSpan`).